### ISSUE TITLE
num_epochs does not work in modified cifar10

### ISSUE THREAD

**Issue by tengerye** [tensorflow/tensorflow#3152]
I am running cifar10 examples and have made some changes. I would like to evaluate every example only once, while the original example can only do mass evaluation. My question are described [here](http://stackoverflow.com/questions/38125436/tensorflow-how-to-evaluate-all-test-set-with-every-example-once-and-only-once). And was suggested by changing function 'tf.train.string_input_producer'.

But it failed and I also found a similar [question](http://stackoverflow.com/questions/35674073/compute-status-not-found-tensor-name-input-producer-limit-epochs-epochs-not) but with no final solution given, @mrry .

And this problem can also solve other problems including this [one](http://stackoverflow.com/questions/36960457/tensorflow-evaluate-with-confusion-matrix) here.

Thank you in advance.

**Comment by tengerye** [comment]
Dear All,
I managed to solve it at the cost that I have to set every thread reads only one test example. I think the queues and threading mechanism of TensorFlow should re-design: the threading should be able to read the rest of queues automatically.

Thank you very much. And I would also appreciated if better solution to be given

**Comment by aselle** [comment]
What version of tensorflow are you using on what operating system and version? Have you tried it with r0.9?

Original Error was
 with NotFoundError: Tensor name "input_producer/limit_epochs/epochs" not found in checkpoint files

**Comment by tengerye** [comment]
I used 0.8 and I believe the problem, that evaluation on each example only once, need redesign. Although in the [post](http://stackoverflow.com/questions/38125436/tensorflow-how-to-evaluate-all-test-set-with-every-example-once-and-only-once), I proposed a go-round solution, but it can only applied to test. For real evaluation, each example should feed and evaluate only once, I still can not find a proper way.
